gm4_enchantment_extractors/data/enchantment_extractors/functions/extract.mcfunction

```diff
@@ -0,0 +1,10 @@
+#@s = item to have enchantments extracted
+#run from enchantment_extractors:process
+
+scoreboard players add extract_chance gm4_ench_random 27
```

Comment:
So, because these are in the extract function, which only gets called once per item, it means that the 27 psudo-random doesn't only gets incremented once per item extracted. This leads to a very not random extraction result. If you get multiple identical items with multiple enchantments, this becomes very clear. Extract the first one, and you'll have decent luck. The second, less so, the third, even less ect... until the score wraps around back to 0 and it starts all over again. This is predictable enough that the players could figure out roughly where in the cycle the score is.

Reply:
This also the case for the general extract chance. I'm not sure why this simple method was used in the previous version. @SpecialBuilder32 Maybe a more complicated LCG should be used?

Reply:
The previous version ran the 27-randomiser constantly when there was an item on an enchantment extractor, meaning that when the value it stopped on depended on how long the item was left on the dropper. Since the player has no way of knowing when the module pulses, how many increments of the randomizer had passed was unknowable.